src/plugins/charts/public/plugin.ts

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { Plugin, CoreSetup } from 'kibana/public';
+
+import { colorMapsService, ThemeService, ColorsService } from './services';
+
+/** @public */
+export type ColorMapsSetup = typeof colorMapsService;
+
+/** @public */
+export type ColorMapsStart = ColorMapsSetup;
+
+export type Theme = Omit<ThemeService, 'init'>;
+export type Color = Omit<ColorsService, 'init'>;
+
+/** @public */
+export interface ChartsPluginSetup {
+  colorMaps: ColorMapsSetup;
+  colors: Color;
+  theme: Theme;
+}
+
+/** @public */
+export type ChartsPluginStart = ChartsPluginSetup;
```

Comment:
Is it necessary to expose the same interface for start and setup from this plugin? This _seems_ like something that would most naturally fit into just setup.

Also, if we just expose at setup time we can remove the need to check for `init`ed services and remove `init` step entirely. Not a blocker though.

Reply:
I don't think it's necessary long term but it was currently being used in both so I resorted to the init approach over using `getStartServices`.